### PR TITLE
WithdrawCommand (and index based commands) Enhancements

### PR DESCRIPTION
**Pull request by uniqly**
* More specific error messages relating to index (specifically size and ambiguous index)
* Users are now informed about `Status` being set to `CLOSED`

Addresses #152 and Addresses #153

### CODE REVIEW

File: src/main/java/seedu/internbuddy/logic/commands/WithdrawCommand.java [hunk 1]
```diff
@@ -70,7 +72,13 @@ public CommandResult execute(Model model) throws CommandException {
...
 
         model.setCompany(companyToEdit, editedCompany);
-        return new CommandResult(String.format(MESSAGE_WITHDRAW_APPLICATION_SUCCESS, applicationToWithdraw));
+        String successMessage = String.format(MESSAGE_WITHDRAW_APPLICATION_SUCCESS,
+                Messages.format(applicationToWithdraw), companyToEdit.getName());
+        if (applicationList.isEmpty()) {
+            return new CommandResult(successMessage + "\nNo more applications for "
+                    + companyToEdit.getName().fullName + ". Status has been set to CLOSED.");
+        }
+        return new CommandResult(successMessage);
```
Comment (blanklogic) [inline src/main/java/seedu/internbuddy/logic/commands/WithdrawCommand.java:81]: Love the new feedback message provided to our users, prompting them about the lack of active applications in the target company after the current command execution ends, thus changing the status to CLOSED.

**Review by blanklogic — APPROVED** [review]
LGTM!

File: src/main/java/seedu/internbuddy/model/application/Application.java [hunk 2]
```diff
@@ -49,13 +50,11 @@ public Application setAppStatus(AppStatus newAppStatus) {
...
-                .append("; Description: ")
-                .append(getDescription())
-                .append("; App Status: ")
-                .append(getAppStatus());
-        return builder.toString();
+        return new ToStringBuilder(this)
+                .add("name", name)
+                .add("description", description)
+                .add("appStatus", appStatus)
+                .toString();
```
Comment (realqijun) [inline src/main/java/seedu/internbuddy/model/application/Application.java:57]: Cool! Shorter code 🧑‍💻🤙

**Review by realqijun — APPROVED** [review]
Great job on unifying the invalid index messages. LGTM

**Pull request merged**